src/scripts/scaling/scalingAction.sh

```diff
@@ -16,10 +14,328 @@ scaling_size=1
+      print j[index]["replicas"]
+INPUT
+  num_ms=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+
+  if [ "${num_ms}" == "null" ]; then
+    num_ms=0
+  fi
+
+  echo "$num_ms"
+}
+
+# Gets the replica count at the Domain level
+function get_num_ms_domain_scope() {
+  local num_ms
+  if [ -x "$(command -v jq)" ]; then
+    num_ms=$(echo "${DOMAIN}" | jq -r '.items[].spec.replicas' )
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+for i in json.load(sys.stdin)["items"]:
+  print i["spec"]["replicas"]
+INPUT
+  num_ms=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+
+  if [ "${num_ms}" == "null" ]; then
+    num_ms=0
+  fi
+
+  echo "$num_ms"
+}
+
+#
+# Function to get minimum replica count for cluster
+# $1 - Domain resource in json format
+# $2 - Name of the cluster
+# $3 - Return value containing minimum replica count
+#
+function get_min_replicas {
+  local domainJson=$1
+  local clusterName=$2
+  local __result=$3
+
+  eval $__result=0
+  minReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .minimumReplicas"
+  minReplicas=$(echo ${domainJson} | jq "${minReplicaCmd}")
+  eval $__result=${minReplicas}
+}
+
+# Get the current replica count for the WLS cluster if defined in the CRD's Cluster
+# configuration.  If WLS cluster is not defined in the CRD then return the Domain
+# scoped replica value, if present.  Returns replica count = 0 if no replica count found.
+# args:
+# $1 "True" if WLS cluster configuration defined in CRD, "False" otherwise
+# $2 Custom Resource Domain
+function get_replica_count() {
+  local in_cluster_startup="$1"
+  local DOMAIN="$2"
+  local num_ms
+  if [ "$in_cluster_startup" == "True" ]
+  then
+    trace "$wls_cluster_name defined in clusters"
+    num_ms=$(get_num_ms_in_cluster "$DOMAIN")
+  else
+    trace "$wls_cluster_name NOT defined in clusters"
+    num_ms=$(get_num_ms_domain_scope "$DOMAIN")
+  fi
+  echo "$num_ms"
+}
+
+# Determine the nuber of managed servers to scale
+# args:
+# $1 scaling action (scaleUp or scaleDown)
+# $2 current replica count
+# $3 scaling increment value
+function calculate_new_ms_count() {
+  local scaling_action="$1"
+  local current_replica_count="$2"
+  local scaling_size="$3"
+  local new_ms
+  if [ "$scaling_action" == "scaleUp" ];
+  then
+    # Scale up by specified scaling size
+    # shellcheck disable=SC2004
+    new_ms=$(($current_replica_count + $scaling_size))
+  else
+    # Scale down by specified scaling size
+    new_ms=$(($current_replica_count - $scaling_size))
+  fi
+  echo "$new_ms"
+}
+
+# Verify if requested managed server scaling count is less than the configured
+# minimum replica count for the cluster.
+# args:
+# $1 Managed server count
+# $2 Custom Resource Domain
+# $3 Cluster name
+function verify_minimum_ms_count_for_cluster() {
+  local new_ms="$1"
+  local domainJson="$2"
+  local clusterName="$3"
+  # check if replica count is less than minimum replicas
+  getMinReplicas "${domainJson}" "${clusterName}" minReplicas
```

Comment:
I assume this should be "get_min_replicas".

Reply:
Great catch!  Thank you